qiskit/circuit/quantumcircuit.py

```diff
@@ -456,12 +456,33 @@ def reverse_bits(self) -> "QuantumCircuit":
             q_1: ┤ H ├─────■──────
                  └───┘
         """
+
+        qreg_bits = {bit for reg in self.qregs for bit in reg}
+        creg_bits = {bit for reg in self.cregs for bit in reg}
+        regless_qubits = []
+        regless_clbits = []
+
+        if set(self.qubits) != qreg_bits:
+            regless_qubits = [bit for bit in self.qubits if bit not in qreg_bits]
+
+        if set(self.clbits) != creg_bits:
+            regless_clbits = [bit for bit in self.clbits if bit not in creg_bits]
+
         circ = QuantumCircuit(
             *reversed(self.qregs),
             *reversed(self.cregs),
             name=self.name,
             global_phase=self.global_phase,
         )
+
+        for qubit in reversed(regless_qubits):
+            circ.qubits.insert(list(reversed(self.qubits)).index(qubit), qubit)
+            circ._qubit_set.add(qubit)
+
+        for clbit in reversed(regless_clbits):
+            circ.clbits.insert(list(reversed(self.clbits)).index(clbit), clbit)
+            circ._clbit_set.add(clbit)
+
```

Comment:
Thanks for your comments @jakelishman and pointing out that runtime complexity issue with my implementation. I agree, it feels like loose qubits weren't even considered when the `reverse_bits()` function was originally written. As @1ucian0 pointed out, even the function description explicitly says it only reverses bits within registers. 

I understood the line `this method is useful for converting a circuit written in little-endian convention to the big-endian equivalent, and vice versa` [from the docs](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.reverse_bits.html#qiskit.circuit.QuantumCircuit.reverse_bits) to be the overall goal of the function. This is what I tried to convey in my [examples/question I posed earlier](https://github.com/Qiskit/qiskit-terra/pull/6762#issuecomment-887641707). 

I'd also like to hear what @ajavadia expects from this function with loose qubits (aside from not erroring out, of course).

Reply:
Yeah, my problem is that the documentation and the original behaviour don't (to me) seem to agree, so I'm finding it hard to work out what's intended - not your fault at all, I think it's more just that Terra programming style/practice moved on a bit since the function was written, and it doesn't line up the same way any more.